server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/ICSAttachmentWorkflowTest.java

```diff
@@ -823,4 +832,50 @@ public void pipelineShouldSendSeveralJSONOverRabbitMQWhenSeveralAttachments() th
         assertThat(amqpRule.readContent()).isEmpty();
     }
 
+    @Test
+    public void mailShouldNotContainCalendarContentInTextBodyButAttachment() throws Exception {
+        Mail mail = FakeMail.builder()
+            .mimeMessage(calendarMessage)
+            .sender(new MailAddress(FROM))
+            .recipient(new MailAddress(RECIPIENT))
+            .build();
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessage(mail);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(RECIPIENT, PASSWORD));
+
+            String receivedMessage = imapMessageReader.readFirstMessageInInbox(RECIPIENT, PASSWORD);
+
+            assertThat(receivedMessage).containsSequence("Content-Type: multipart/mixed", "Content-Disposition: attachment");
+        }
+    }
+
+    @Test
+    public void mailShouldKeepAllHeadersWhenConvertingFromTextCalendarToAttachment() throws Exception {
+        List<Header> allHeaders = Collections.list(calendarMessage.getAllHeaders());
+        List<String> allStringHeaders = allHeaders.stream()
+            .map(Header::getName)
+            .collect(Guavate.toImmutableList());
+
+        Mail mail = FakeMail.builder()
+            .mimeMessage(calendarMessage)
+            .sender(new MailAddress(FROM))
+            .recipient(new MailAddress(RECIPIENT))
+            .build();
+
+        try (SMTPMessageSender messageSender = SMTPMessageSender.noAuthentication(LOCALHOST_IP, SMTP_PORT, JAMES_APACHE_ORG);
+             IMAPMessageReader imapMessageReader = new IMAPMessageReader(LOCALHOST_IP, IMAP_PORT)) {
+            messageSender.sendMessage(mail);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(messageSender::messageHasBeenSent);
+            calmlyAwait.atMost(Duration.ONE_MINUTE).until(() -> imapMessageReader.userReceivedMessage(RECIPIENT, PASSWORD));
+
+            String messageHeaders = imapMessageReader.readFirstMessageHeadersInInbox(RECIPIENT, PASSWORD);
+
+            for (String header : allStringHeaders) {
+                assertThat(messageHeaders).contains(header);
```

Comment:
This maybe do not need to be integration test. Maybe unit test is easier.

And you are just checking header names. It's not enough.

Reply:
Removed